packages/tenant-process/src/services/tenantService.ts

```diff
@@ -515,6 +515,76 @@ export function tenantServiceBuilder(
       return updatedTenant;
     },
 
+    async internalAssignCertifiedAttribute(
+      {
+        tenantOrigin,
+        tenantExternalId,
+        attributeOrigin,
+        attributeExternalId,
+        correlationId,
+      }: {
+        tenantOrigin: string;
+        tenantExternalId: string;
+        attributeOrigin: string;
+        attributeExternalId: string;
+        correlationId: string;
+      },
+      logger: Logger
+    ): Promise<void> {
+      logger.info(
+        `Assigning certified attribute (${attributeOrigin}/${attributeExternalId}) to tenant (${tenantOrigin}/${tenantExternalId})`
+      );
+
+      const tenantToModify = await readModelService.getTenantByExternalId({
+        origin: tenantOrigin,
+        value: tenantExternalId,
+      });
+
+      assertTenantExists(
+        unsafeBrandId(`${tenantOrigin}/${tenantExternalId}`),
+        tenantToModify
+      );
+
+      const attributeToAssign =
+        await readModelService.getAttributeByOriginAndCode({
+          origin: attributeOrigin,
+          code: attributeExternalId,
+        });
+
+      if (!attributeToAssign) {
+        throw attributeNotFound(`${attributeOrigin}/${attributeExternalId}`);
+      }
+
+      const tenantWithNewAttribute = await assignCertifiedAttribute({
+        targetTenant: tenantToModify.data,
+        attribute: attributeToAssign,
+      });
+
+      const tenantCertifiedAttributeAssignedEvent =
+        toCreateEventTenantCertifiedAttributeAssigned(
+          tenantToModify.metadata.version,
+          tenantWithNewAttribute,
+          attributeToAssign.id,
+          correlationId
+        );
+
+      const [, tenantKindUpdatedEvent] = await reevaluateTenantKind({
+        tenant: tenantWithNewAttribute,
+        version: tenantToModify.metadata.version + 1,
+        readModelService,
+        correlationId,
+      });
```

Comment:
In my opinion it's better to extract the `toCreateEventTenantKindUpdated` from the function `reevaluateTenantKind` and put it here, so it's always easy to detect which event is being generated and what tenant is in it.

Reply:
I like the reusability of this function because it reduces the proliferation of bugs, but the name of the function is misleading and one of its return parameters is also unused here and this could mean that `reevaluteTenantKind` should be removed in favor of custom logic.
So I agree whit @taglioni-r.

Reply:
Sounds good to me, the important parts are to have functions about what they say and that the reusability is real.